### ISSUE TITLE
About Cross-modal ensembles

### ISSUE THREAD

**Issue by mishimario** [linzhiqiu/cross_modal_adaptation#9]
Hi, I have a question about Cross-modal ensembles. I read the paper, but I couldn't find the part of Cross-modal ensembles in this repository. So I guess it's not used in this codes. Because also it uses cross-entropy loss not softmax loss. Do you know something? Thank you in advance!

**Comment by linzhiqiu** [comment]
Hey,

In our paper, we use softmax loss to mean cross-entropy loss, so they are the same thing.

About "Cross-modal ensembles." in our paper, it is a theoretical justification of our method, so it has nothing to do with implementation. However, you can still find WiSE-FT (Robust-Finetuning of Zero-Shot Models) code in our codebase as we directly compare against them.

Fun fact: if you calculate the gradient wrt to the cross-entropy loss (softmax loss) in linear settings (the setting of our cross-modal linear-probing), you can find that it is always a linear combination of the input image and text features of this class.

**Comment by mishimario** [comment]
I got it! That's a theoretical explanation of the final shared layer! Thank you very much!
